### ISSUE TITLE
error in extrusion + non_conforming

### ISSUE THREAD

**Issue by alessiofumagalli** [pmgbergen/porepy#75]
Hi, I have a problem when I couple the extrusion and the non-conforming algorithm. However, I think the main problem is due to the latter. Note: if considering the extrusion couple with a conforming algorithm it works.
Minimal example:

```python
import numpy as np                                                               
import scipy.sparse as sps                                                       
                                                                                 
from porepy.viz.exporter import Exporter                                         
from porepy.fracs import importer                                                
                                                                                 
from porepy.fracs import extrusion                                               
from porepy.fracs.fractures import Fracture, FractureNetwork                     
from porepy.fracs import meshing                                                 
                                                                                                                                                               
folder_export = 'example_4_vem/'                                                 
file_export = 'vem'                                                              
tol = 1e-8                                                                       
                                                                                 
# The fractures are not aligned with the mesh                                    
pts = np.array([[0, 0], [2, 2], [1, 1], [1, 3]]).T                               
edges = np.array([[0, 1], [2, 3]]).T                                             
                                                                                 
extrustion_kwargs = {}                                                           
extrustion_kwargs['tol'] = tol                                                   
extrustion_kwargs['exposure_angle'] = 2*np.pi/3.*np.ones(edges.shape[1])         
                                                                                 
fractures = extrusion.fractures_from_outcrop(pts, edges, **extrustion_kwargs)    
network = FractureNetwork(fractures, tol=tol)                                    
                                                                                 
# If uncomment the code work with the new network                                
# f_1 = Fracture(np.array([[-1, -1, 0], [1, -1, 0], [1, 1, 0], [-1, 1, 0]]).T)   
# f_2 = Fracture(np.array([[-1, 0, -1], [1, 0, -1], [1, 0, 1], [-1, 0, 1]]).T)   
# f_3 = Fracture(np.array([[-0.75, 0.25, -0.75], [0.75, 0.25, -0.75], [0.75,     
#                            0.25, 0.75], [-0.75, 0.25, 0.75]]).T)               
# network = FractureNetwork([f_1, f_2, f_3])                                     
                                                                                 
mesh_kwargs = {}                                                                 
mesh_kwargs['mesh_size'] = {'mode': 'weighted',                                  
                            'value': 0.5,                                        
                            'bound_value': 0.5,                                  
                            'tol': tol}                                          
                                                                                 
gb = meshing.dfn(network, conforming=False, **mesh_kwargs)                       
                                                                                 
gb.remove_nodes(lambda g: g.dim == 0)                                            
gb.compute_geometry()                                                            
gb.assign_node_ordering()                                                        
                                                                                 
save = Exporter(gb, file_export, folder_export)                                  
save.write_vtk()
```

**Comment by keileg** [comment]
Update: There was a bug in the computation of mesh sizes in FractureNetwork. Should be fixed by 84f6430 . Also note that this relates to the old (to be deleted) computation of mesh sizes.

**Comment by alessiofumagalli** [comment]
Thanks @keileg for the clarifications. I think it's important also to consider the Y intersections, and it'll be on me. But we need to discuss when you'll have time.

About the new way of computing the mesh size, is the new way compatible also with the "weighted" algorithm developed by @IvarStefansson and myself?

**Comment by keileg** [comment]
Actually, gluing the meshes should not be too difficult. To create the missing 1d grid (in the abutting fracture): 
1. Find the points that lies in / close to the plane of the other fracture, probably by using cg.dist_points_polygon (checks: a) the point we found should be on a line, b) they should be in the interval of the intersection),
2. Sort the points along the line, define a 1d TensorGrid, insert in the right place in the lists, and move on.
To know where to generate such a grid, we can use FractureNetwork.Intersection, which stores information as to whether the intersection is on the boundary of two fractures. If True, an auxiliary mesh is needed. 

If this works as I picture it, it should become not a hack, but a part of the regular workflow for DFNs. I will create a branch and start coding, then we can discuss progress next week.


Compatibility of meshing: In spirit, but not in terms of keywords. The idea is that you specify an ideal and minimum mesh size (as parameters assigned to points in gmsh), with keywords h_min and h_ideal, see fracs.simplex.tetrahedral_grid, no examples available yet. I will change keywords, and do several other minor improvements, in the not too distant future.

**Comment by keileg** [comment]
Close this when merge of dfn_y_intersect is accepted.

**Comment by keileg** [comment]
Fixed  by #93 .
